ios/browser/api/certificate/brave_certificate.mm

```diff
@@ -0,0 +1,189 @@
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#import "brave/ios/browser/api/certificate/brave_certificate.h"
+#import "brave/ios/browser/api/certificate/common/brave_certificate_enums.h"
+#import "brave/ios/browser/api/certificate/models/brave_certificate_fingerprint.h"
+#import "brave/ios/browser/api/certificate/models/brave_certificate_model_private.h"
+#import "brave/ios/browser/api/certificate/models/brave_certificate_public_key_info.h"
+#import "brave/ios/browser/api/certificate/models/brave_certificate_rdns_sequence.h"
+#import "brave/ios/browser/api/certificate/models/brave_certificate_signature.h"
+#import "brave/ios/browser/api/certificate/utils/brave_certificate_utils.h"
+#import "brave/ios/browser/api/certificate/utils/brave_certificate_x509_utils.h"
+
+// base
+#include "base/logging.h"
+#include "base/memory/ref_counted.h"
+#include "base/strings/string_piece.h"
+#include "base/strings/sys_string_conversions.h"
+#include "base/time/time.h"
+
+// net
+#include "net/base/net_export.h"
+#include "net/cert/internal/cert_errors.h"
+#include "net/cert/internal/parsed_certificate.h"
+#include "net/cert/internal/verify_name_match.h"
+#include "net/cert/internal/verify_signed_data.h"
+#include "net/cert/x509_cert_types.h"
+#include "net/cert/x509_certificate.h"
+#include "net/cert/x509_util.h"
+#include "net/cert/x509_util_ios.h"
+#include "net/der/input.h"
+
+// third_party
+#include "third_party/boringssl/src/include/openssl/pool.h"
+
+#if !defined(__has_feature) || !__has_feature(objc_arc)
+#error "This file requires ARC support."
+#endif
+
+// MARK: - Implementation
+
+@interface BraveCertificateModel () {
+  base::ScopedCFTypeRef<CFDataRef> cert_data_;
+  scoped_refptr<net::ParsedCertificate> extended_cert_;
+  base::ScopedCFTypeRef<SecKeyRef> public_key_;
+}
+@end
+
+@implementation BraveCertificateModel
+- (nullable instancetype)initWithCertificate:(SecCertificateRef)certificate {
+  if ((self = [super init])) {
+    cert_data_ =
+        base::ScopedCFTypeRef<CFDataRef>(SecCertificateCopyData(certificate));
+    if (!cert_data_) {
+      return nullptr;
+    }
+
+    public_key_ =
+        base::ScopedCFTypeRef<SecKeyRef>(SecCertificateCopyKey(certificate));
+
+    bssl::UniquePtr<CRYPTO_BUFFER> cert_buffer(
+        net::X509Certificate::CreateCertBufferFromBytes(
+            reinterpret_cast<const char*>(CFDataGetBytePtr(cert_data_)),
+            CFDataGetLength(cert_data_)));
+
+    if (!cert_buffer) {
+      return nullptr;
+    }
+
+    net::CertErrors errors;
+    extended_cert_ =
+        scoped_refptr<net::ParsedCertificate>(net::ParsedCertificate::Create(
+            std::move(cert_buffer),
+            net::x509_util::DefaultParseCertificateOptions() /* {} */,
+            &errors));
+
+    if (!extended_cert_) {
+      LOG(ERROR) << errors.ToDebugString();
```

Comment:
Can you verify this logs on brave-ios side? We've used `VLOG` in other places, but I'm not sure I've had anything use plain `LOG`